lib/active_model_serializers/adapter/json_api/pagination_links.rb

```diff
@@ -35,19 +35,21 @@ def as_json
         private
 
         def pages_from
-          return {} if collection.total_pages <= FIRST_PAGE
-
           {}.tap do |pages|
-            pages[:self] = collection.current_page
+            pages[:self]  = collection.current_page
+            pages[:first] = FIRST_PAGE
+            pages[:last]  = collection.total_pages
 
-            unless collection.current_page == FIRST_PAGE
-              pages[:first] = FIRST_PAGE
-              pages[:prev]  = collection.current_page - FIRST_PAGE
-            end
+            if collection.total_pages > 0
+              unless collection.current_page == FIRST_PAGE
+                pages[:prev] = collection.current_page - FIRST_PAGE
+              end
 
-            unless collection.current_page == collection.total_pages
-              pages[:next] = collection.current_page + FIRST_PAGE
-              pages[:last] = collection.total_pages
+              unless collection.current_page == collection.total_pages
+                pages[:next] = collection.current_page + FIRST_PAGE
+              end
+            else
+              pages[:last] = FIRST_PAGE
             end
           end
```

Comment:
sure, though it might be nice to refactor to something like

```ruby
  def as_json
      {
        "self": location_url,
        "first": first_page_url,
        "prev": prev_page_url,
        "next": next_page_url,
        "last": last_page_url
      }
    end

    def location_url
      url_for_page(collection.current_page)
    end

    def first_page_url
      url_for_page(1)
    end

    def prev_page_url
      return nil if collection.first_page?
      url_for_page(collection.prev_page)
    end

    def next_page_url
      return nil if collection.last_page? || collection.out_of_range?
      url_for_page(collection.next_page)
    end

    def last_page_url
      url_for_page(collection.total_pages)
    end

    def url_for_page(number)
      params = query_parameters.dup
      params[:page] = { page: per_page, number: number }
      context.url_for(action: :index, params: params)
    end

    def per_page
       @per_page ||= collection.try(:per_page) || collection.try(:limit_value) || collection.size
    end
```